frontend/src/__tests__/PostDialog.test.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import PostDialog from '../components/PostDialog';
+
+const mockProject = {
+  project_name: 'AI Research',
+  description: 'Exploring AI in education.',
+  desired_qualifications: 'Experience in Python and AI frameworks.',
+  umbrella_topics: ['AI', 'Education'],
+  research_periods: ['Spring 2024', 'Fall 2024'],
+  is_active: true,
+  majors: ['Computer Science', 'Education'],
+  faculty: {
+    first_name: 'John',
+    last_name: 'Doe',
+    email: 'john.doe@sandiego.edu',
+  },
+};
+
+describe('PostDialog Component', () => {
+  it('renders the project details correctly', () => {
+    render(<PostDialog open={true} onClose={() => {}} project={mockProject} />);
```

Comment:
There is a better way to render the PostDialog component by removing the 'open' prop. Please remove it so we have less props to pass in. We don't want to pass in a prop here because we want to set it dynamically instead of having to pass in a value.